Neos.Flow/Classes/Configuration/ConfigurationManager.php

```diff
@@ -540,17 +384,32 @@ public function flushConfigurationCache()
         }
     }
 
+    /**
+     * Generate configuration with environment variables replaced without modifying or loading the cache
+     *
+     * @param string $configurationType The kind of configuration to fetch
+     */
+    protected function processConfigurationType(string $configurationType)
+    {
+        if ($this->temporaryDirectoryPath !== null) {
+            $cachePathAndFilename = $this->constructConfigurationCachePath().'.tmp';
```

Comment:
Hmm, what about using something like
```
$cachePathAndFilename = $this->constructConfigurationCachePath() . '.' . uniqid($configurationType, true) . '.tmp';
```
But this also means, the `$configurationType` needs to be sanitized to allowed filesystem characters. Maybe `uniqid(crc32($configurationType))` could be enough?

Reply:
sounds fine to me :)

Reply:
I've fixed concurrency using `uniqid` and `crc32`.

The test, that is now failing is because of the recent changes in the master I merged - I think.